CMakeLists.txt

```diff
@@ -68,6 +70,22 @@ endif(CPPDAP_BUILD_TESTS)
 ###########################################################
 # JSON library
 ###########################################################
+
+if(CPPDAP_USE_EXTERNAL_NLOHMANN_JSON_PACKAGE AND CPPDAP_USE_EXTERNAL_RAPIDJSON_PACKAGE)
+    message(FATAL_ERROR "Don't set both CPPDAP_USE_EXTERNAL_NLOHMANN_JSON_PACKAGE and CPPDAP_USE_EXTERNAL_RAPIDJSON_PACKAGE")
+elseif(CPPDAP_USE_EXTERNAL_NLOHMANN_JSON_PACKAGE OR CPPDAP_USE_EXTERNAL_RAPIDJSON_PACKAGE)
+    set(USING_EXTERNAL_JSON true)
+    if(CPPDAP_USE_EXTERNAL_NLOHMANN_JSON_PACKAGE)
+        find_package(nlohmann_json CONFIG REQUIRED)
+        set(CPPDAP_JSON_LIBRARY "nlohmann")
+    elseif(CPPDAP_USE_EXTERNAL_RAPIDJSON_PACKAGE)
+        find_package(RapidJSON CONFIG REQUIRED)
+        set(CPPDAP_JSON_LIBRARY "rapid")
+    endif()
+else()
+    set(USING_EXTERNAL_JSON false)
+endif()
+
```

Comment:
heh, looking at the surrounding code I see that L100: `message(FATAL_ERROR "Could not determine JSON library from ${CPPDAP_JSON_LIBRARY}")` should be `message(FATAL_ERROR "Could not determine JSON library from ${CPPDAP_JSON_DIR}")`

Reply:
Fixed